Cargo.toml

```diff
@@ -46,3 +46,16 @@ tokio-stream = { version = "0.1" }
 tonic = { version = "0.12" }
 tracing = { version = "0.1" }
 tracing-subscriber = { version = "0.3", features = ["fmt",  "json",  "env-filter"] }
+
+# Lints are set to warn for development, which are promoted to errors in CI.
+[workspace.lints.clippy]
+# Pedantic lints are set to a lower priority which allows lints in the group to be selectively enabled.
+pedantic = { level = "warn", priority = -1 }
+cast_possible_truncation = "allow"     # Overly many instances especially regarding indices.
+large_types_passed_by_value = "allow"  # Triggered by BlockHeader being Copy + 334 bytes.
```

Comment:
This was somewhat interesting. `BlockHeader` is `Copy`, but is also quite large at 334 bytes.

This combination causes clippy to suggest passing it by reference even though the function requires by value. The assumption being that the reference will be copied.

I don't quite agree with clippy here, but it did make me wonder whether we should remove `Copy` from `BlockHeader`?

Reply:
https://github.com/0xPolygonMiden/miden-base/issues/1076

Reply:
This did show me a lint configuration `avoid-breaking-exported-api` which we should probably disable. I'll do that in a separate PR though.